audbcards/core/datacard.py

```diff
@@ -161,83 +147,143 @@ def file_duration_distribution(self) -> str:
+        is not ``None``,
+        an example media file is cached in the folder
+        ``<dataset-name>-<dataset-version>-player-media/``
+        inside
+        ``<cache-root>/<dataset-name>/<dataset-version>/``,
+        using the same sub-folder structure
+        as the media file has inside its dataset.
+        If :attr:`audbcards.Datacard.sphinx_build_dir`
+        is not ``None``,
+        the media sub-folder structure
+        is also copied
+        to the sphinx build dir into
+        ``<sphinx-build-dir>/<path>/<dataset-name>/``.
+
+        If :attr:`audbcards.Datacard.sphinx_src_dir` is not ``None``,
+        a plot of the waveform of the media file
+        is cached under
+        ``<dataset-name>-<dataset-version>-player-waveform.png``
+        inside
+        ``<cache-root>/<dataset-name>/<dataset-version>/``.
+        It is also copied to the sphinx source folder into
+        ``<sphinx-src-dir>/<path>/<dataset-name>/``.
 
-        """
-        if file is None:
-            file = self.example_media
+        Returns:
+            String containing RST code to include the player
 
-        # use audb cache instead of dataset.cache_root
-        media_src_dir = (
-            f"{audb.default_cache_root()}/"
-            f"{audb.flavor_path(self.dataset.name, self.dataset.version)}"
+        """
+        # Cache is organized as `<cache_root>/<name>/<version>/`
+        cache_folder = audeer.path(
+            self.cache_root,
+            self.dataset.name,
+            self.dataset.version,
+        )
+        cache_example_media = audeer.path(
+            cache_folder,
+            f"{self.dataset.name}-{self.dataset.version}-player-media",
+            self.dataset.example_media,
+        )
+        cache_waveform_file = audeer.path(
+            cache_folder,
+            f"{self.dataset.name}-{self.dataset.version}-player-waveform.png",
         )
 
-        # Move file to build folder
-        if self.sphinx_build_dir is not None:
-            media_dst_dir = audeer.path(
-                self.sphinx_build_dir,
+        def load_media_to_cache(cache_example_media: str):
+            r"""Load media file with audb and copy to audbcards cache.
+
+            Args:
+                cache_example_media: full path to media file in cache
+
+            """
+            media = audb.load_media(
+                self.dataset.name,
+                self.dataset.example_media,
+                version=self.dataset.version,
+                verbose=False,
+            )[0]
+            audeer.mkdir(os.path.dirname(cache_example_media))
+            shutil.copy(media, cache_example_media)
+
+        # Add plot of waveform
+        if self.sphinx_src_dir is not None:
+            if not os.path.exists(cache_example_media):
+                load_media_to_cache(cache_example_media)
+
+            if not os.path.exists(cache_waveform_file):
+                signal, sampling_rate = audiofile.read(
```

Comment:
The functionality to fetch audio examples and copy them to the cache location is factored into a separate function `load_media_cache`. Would it make sense to keep these bits parallel, i.e. have a dedicated plotter function?

Reply:
Yes, this makes it indeed easier to understand. I added now a function called `plot_waveform_to_cache()`.

In order to integrate the setting of the cache folder inside that function, I also had to change what is included in the returned `player_str`. Before we always returned the same string, independent if the files were accessible or not.
I think, that was anyway not a good approach. Now I return an empty string if no file is there (when `self.sphinx_src_dir` and `self.sphinx_build_dir` are both `None`), and add to the string on the fly.

Reply:
To me the new implementation makes sense and is a lot easier to read.